zookeeper-server/src/main/java/org/apache/zookeeper/server/ObserverBean.java

```diff
@@ -49,10 +50,11 @@ public String getQuorumAddress() {
 
     public String getLearnerMaster() {
         QuorumPeer.QuorumServer learnerMaster = observer.getCurrentLearnerMaster();
-        if (learnerMaster == null || learnerMaster.addr == null) {
+        InetSocketAddress address = learnerMaster.addr.getReachableOrOne();
+        if (learnerMaster == null || address == null) {
```

Comment:
if `learnerMaster` is `null` here then you'll get a NPE on the `learnerMaster.addr.getReachableOrOne()` above.

Reply:
thanks, nice catch!
(actually beside this, we also got a NoSuchElementException if learnerMaster.addr is empty)
I will fix this in the next commit